conformance/tests/httproute-request-percentage-mirror.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/config"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/conformance/utils/tlog"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+const (
+	concurrentRequests    = 10
+	tolerancePercentage   = 15.0
+	totalRequests         = 500.0
+	numDistributionChecks = 5
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteRequestPercentageMirror)
+}
+
+var HTTPRouteRequestPercentageMirror = suite.ConformanceTest{
+	ShortName:   "HTTPRouteRequestPercentageMirror",
+	Description: "An HTTPRoute with percentage based request mirroring",
+	Manifests:   []string{"tests/httproute-request-percentage-mirror.yaml"},
+	Features: []features.FeatureName{
+		features.SupportGateway,
+		features.SupportHTTPRoute,
+		features.SupportHTTPRouteRequestPercentageMirror,
+	},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		var (
+			ns      = "gateway-conformance-infra"
+			routeNN = types.NamespacedName{Name: "request-percentage-mirror", Namespace: ns}
+			gwNN    = types.NamespacedName{Name: "same-namespace", Namespace: ns}
+			gwAddr  = kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		)
+
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		// TODO(liorlieberman) add another test to show fraction takes precedence over percent
```

Comment:
Reminder not to forget this

Reply:
Can you really set both fraction and percent at the same time? That seems like a bug if possible?

Reply:
I was following the GEP - https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-3171/index.md?plain=1#L88-L92

I can see now that its not up to date, and the API that was merged has a CEL to validate that you cant configure both. (https://github.com/kubernetes-sigs/gateway-api/blob/99a3934c6bc1ce0874f3a4c5f20cafd8977ffcb4/apis/v1/httproute_types.go#L858)

Removing it